python/sglang/srt/mem_cache/radix_cache.py

```diff
@@ -461,23 +461,47 @@ def _collect_leaves(self):
         return ret_list
 
     def _record_store_event(self, node: TreeNode):
+        # Emit one BlockStored event per fixed-size "page" so that each
+        # event always represents exactly ``self.page_size`` contiguous tokens.
+        # This keeps the wire contract simple for downstream consumers like
+        # Dynamo that assume uniform block sizes.
         if self.enable_kv_cache_events:
-            block_hash = hash(tuple(node.key))
-            parent_block_hash = hash(tuple(node.parent.key))
-            self.kv_event_queue.append(
-                BlockStored(
-                    block_hashes=[block_hash],
-                    parent_block_hash=parent_block_hash,
-                    token_ids=node.key,
-                    block_size=len(node.key),
-                    lora_id=None,
-                )
+            # The first page inherits its parent from the radix tree; each
+            # subsequent page chains off the previous page we just emitted so
+            # that the receiver can reconstruct the original order.
+            parent_block_hash = (
+                hash(tuple(node.parent.key)) if node.parent else None
```

Comment:
If the parent key has multiple pages, dynamo won't have a record of this hash. Should this be the hash of the last page of the parent node instead?